pkg/request/pending.go

```diff
@@ -259,6 +319,29 @@ func (ps *PendingStore) now() time.Time {
 	return ps.lastTick.Load().(time.Time)
 }
 
+// GetAllRequests returns all stored requests in the same order of their arrival, the oldest one will be the first
+func (ps *PendingStore) GetAllRequests(max uint64) [][]byte {
+
+	requests := make([][]byte, 0, max*2)
+
+	ps.bucketsLock.RLock()
+	for _, b := range ps.buckets {
+		b.requests.Range(func(_, v interface{}) bool {
+			requests = append(requests, v.([]byte))
+			return true
+		})
+	}
+	ps.bucketsLock.RUnlock()
```

Comment:
what if right after the lock is released, current bucket gets into the buckets and an empty bucket is created? should we not hold the lock on the buckets throughout the function's execution?

Reply:
ok good idea
will do